react-validated-input-component/my-app/src/ValidateInput.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+
+export default function ValidateInput() {
+  const [inputValue, setInputValue] = useState('');
+
+  const regex = new RegExp(
+    '(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$'
+  );
+
+  let inputValueLength: number = inputValue.length;
+  let xIHidden: string = 'hidden';
+  let checkHidden: string = 'hidden';
+  let shortPassword: string = 'hidden';
+  let noPassword: string = 'hidden';
+  let regexPass: string = 'hidden';
+
+  if (inputValueLength === 0 || inputValueLength < 8) {
+    xIHidden = '';
+  }
+  if (inputValueLength === 0) {
+    noPassword = '';
+  }
+  if (inputValueLength >= 8 && regex.test(inputValue)) {
+    checkHidden = '';
+  }
+  if (inputValueLength < 8 && inputValueLength > 0) {
+    shortPassword = '';
+  }
+  if (inputValueLength > 8 && !regex.test(inputValue)) {
+    regexPass = '';
+  }
```

Comment:
I don't know if it's better code or not, but I would have written these like the following (just showing one example):
```
const regexPass = (inputValueLength > 8 && !regex.test(inputValue)) ? '' : 'hidden';
```